build/linux/rpmDependenciesGenerator.ts

```diff
@@ -0,0 +1,41 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import { spawnSync } from 'child_process';
+import { calculatePackageDeps, mergePackageDeps } from './linux-installer/rpm/rpmDependencyScripts';
+import { additionalDeps } from './linux-installer/rpm/additionalDeps';
+
+export function getRpmDependencies(buildDir: string, applicationName: string): string[] {
+	// Get the files for which we want to find dependencies.
+	const findResult = spawnSync('find', [buildDir, '-name', '*.node']);
+	if (findResult.status) {
+		console.error('Error finding files:');
+		console.error(findResult.stderr.toString());
+		return [];
+	}
+
+	const files = findResult.stdout.toString().trimEnd().split('\n');
+
+	const appPath = `${buildDir}/${applicationName}`;
+	files.push(appPath);
+
+	// Generate the dependencies.
+	const dependencies: Set<string>[] = files.map((file) => calculatePackageDeps(file));
```

Comment:
Require list should remove the shared libraries the runtime already bundles. for ex: https://source.chromium.org/chromium/chromium/src/+/refs/tags/98.0.4758.109:chrome/installer/linux/BUILD.gn;l=64-80

```
libEGL.so,
libffmpeg.so,
libGLESv2.so,
libvk_swiftshader.so,
libvulkan.so.1
```

these are removed via the `shlibs` argument of `calculate_package_deps.py` 

https://source.chromium.org/chromium/chromium/src/+/refs/tags/98.0.4758.109:chrome/installer/linux/BUILD.gn;l=158
https://source.chromium.org/chromium/chromium/src/+/refs/tags/98.0.4758.109:chrome/installer/linux/rpm/calculate_package_deps.py;l=27
https://source.chromium.org/chromium/chromium/src/+/refs/tags/98.0.4758.109:chrome/installer/linux/rpm/calculate_package_deps.py;l=56-58
https://source.chromium.org/chromium/chromium/src/+/refs/tags/98.0.4758.109:chrome/installer/linux/rpm/calculate_package_deps.py;l=67

Similar filtering can be done here after calculating the dependencies.